dv/uvm/core_ibex/riscv_dv_extension/ibex_log_to_trace_csv.py

```diff
@@ -69,17 +70,17 @@ def _process_ibex_sim_log_fd(log_fd, csv_fd, full_trace=True):
             trace_entry.pc = m.group("pc")
             trace_entry.binary = m.group("bin")
             if full_trace:
+                trace_entry.operand = m.group("instr").split()[1]
+                process_trace(trace_entry)
                 # Convert the operands into ABI format for
                 # the functional coverage flow
-                operands = m.group("instr").split()[1]
-                trace_entry.operand = convert_operands_to_abi(operands)
-                process_trace(trace_entry)
+                trace_entry.operand = convert_operands_to_abi(trace_entry.operand)
 
-        c = RD_RE.search(line)
-        if c:
-            trace_entry.gpr.append('{}:{}'
-                                   .format(gpr_to_abi("x%0s" % c.group("rd")),
-                                           c.group("rd_val")))
+            c = RD_RE.search(line)
+            if c:
+                trace_entry.gpr.append('{}:{}'
+                                       .format(gpr_to_abi("x%0s" % c.group("rd")),
+                                               c.group("rd_val")))
```

Comment:
The extra 4 char indent pushes this over 80 chars, which is why I noticed it. Maybe it would be better to use some local variables to avoid the deep nesting?
```python
            c = RD_RE.search(line)
            if c:
                abi_name = gpr_to_abi('x{}'.format(c.group('rd')))
                gpr_entry = '{}:{}'.format(abi_name, c.group("rd_val"))
                trace_entry.gpr.append(gpr_entry)
```
Same number of lines, but they're shorter and it's more obvious what's going on (I think).

Another thing that might make this easier to read (sort of orthogonal to the above) is to change the check after `INSTR_RE.search` to continue on failure. This drops 4 spaces of indentation, which I think is probably better (and "skip this line if I don't understand it" isn't particularly confusing control flow). With the changes I suggested above, the result ends up looking something like this:
```python
    for line in log_fd:
        if re.search("ecall", line):
            break

        # Extract instruction information
        m = INSTR_RE.search(line)
        if m is None:
            continue

        instr_cnt += 1
        # Write the extracted instruction to a csvcol buffer file
        trace_entry = RiscvInstructionTraceEntry()
        trace_entry.instr_str = m.group("instr")
        trace_entry.instr = m.group("instr").split()[0]
        trace_entry.pc = m.group("pc")
        trace_entry.binary = m.group("bin")
        if full_trace:
            expand_trace_entry(trace_entry, m.group("instr").split()[1])

        c = RD_RE.search(line)
        if c:
            abi_name = gpr_to_abi('x{}'.format(c.group('rd')))
            gpr_entry = '{}:{}'.format(abi_name, c.group("rd_val"))
            trace_entry.gpr.append(gpr_entry)

        trace_csv.write_trace_entry(trace_entry)
```

Reply:
whoops, thanks for catching the line length violation!
I'll restructure this part as well to match your suggested style.